pages/instructorAssessments/instructorAssessments.ejs

```diff
@@ -15,7 +16,41 @@
             trigger: 'hover',
         }).on("show.bs.popover", function() {
             $($(this).data("bs.popover").getTipElement()).css("max-width", "80%");
-        });;
+        });
+
+        var assessment_ids_needing_stats_update = <%- JSON.stringify(assessment_ids_needing_stats_update) %>;
+        async.eachLimit(assessment_ids_needing_stats_update, 3, function(assessment_id, callback) {
+          fetch('<%= urlPrefix %>/instance_admin/assessments/stats/' + assessment_id)
```

Comment:
I experimented with using [htmx](https://htmx.org) to do this lazy-loading (see b22954fc8bd877bf2b5502f405be2836c7955f4d). It was super short and clean, but there was no good way to both fetch the updated rows incrementally and also limit the concurrency of fetches. The code in b22954fc8bd877bf2b5502f405be2836c7955f4d fetched the entire table at once, which allowed us to limit concurrency server-side. This was nice, but in a really big class could still lead to timeouts if the number of assessments became very large. In the end I opted to write the fetching code by hand, which let me limit concurrency client-side and incrementally fetch each row, so no single fetch will take very long.

Reply:
Makes sense. Note that if we're using `fetch`, we can safely use `async`/`await`/`const`/etc.